lang/cs/Org.Apache.REEF.Bridge.Core.Client/Common/Events/FailedJob.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+using System;
+using System.Runtime.Serialization;
+using Org.Apache.REEF.Common.Client;
+using Org.Apache.REEF.Common.Exceptions;
+using Org.Apache.REEF.Utilities;
+
+namespace Org.Apache.REEF.Bridge.Core.Client.Common.Events
+{
+    internal sealed class FailedJob : IFailedJob
+    {
+        public FailedJob(string jobId, string message) : this(jobId, message, null)
+        {
+        }
+
+        public FailedJob(string jobId, string message, byte[] data)
+        {
+            Id = jobId;
+            Message = message;
+            Data = Optional<byte[]>.OfNullable(data);
+        }
+
+        public string Id { get; }
+        public string Message { get; set; }
+        public Optional<string> Description { get; set; }
```

Comment:
I'm only following the API of IFailJob, which I agree needs to be revised, but I am not allowed to do that in this PR since it requires a release to deprecate, and another release to remove, as per REEF restrictions on public API changes; so I was told myself :)